IssueCreator/IssueCreator/IssueManager.cs

```diff
@@ -0,0 +1,270 @@
+﻿using IssueCreator.Models;
+using Microsoft.Extensions.Caching.Memory;
+using Octokit;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using ZenHub;
+
+namespace IssueCreator
+{
+    public class IssueManager
+    {
+        ZenHubClient _zenHubClient;
+        GitHubClient _githubClient;
+        MemoryCache _cache = new MemoryCache(new MemoryCacheOptions());
+
+        public static IssueManager Create(Settings settings)
+        {
+            IssueManager manager = new IssueManager();
+            if (!manager.RefreshGitHubToken(settings.GitHubToken))
+            {
+                return null;
+            }
+
+            manager.RefreshZenHubToken(settings.ZenHubToken);
+
+            return manager;
+        }
+
+        private IssueManager()
+        {
+        }
+
+        public void RefreshZenHubToken(string newToken)
+        {
+            _zenHubClient = new ZenHubClient(newToken);
+        }
+
+        public bool RefreshGitHubToken(string newToken)
+        {
+            try
+            {
+                _githubClient = new GitHubClient(new ProductHeaderValue("GitHubSync"))
```

Comment:
Should that be a constant?  Or configurable?

Reply:
Changed it to IssueCreator -- this is how the app should identify itself to GH. I don't think this should be changed.